### PR TITLE
[tests] Fix build on linux

### PR DESCRIPTION
**Pull request by hudokkow**
This makes tests build again on linux. I still get 6 failing tests but that's work for another time.
cc/ @Razzeee 

EDIT: Don't merge this yet. There's a bunch of warnings I want to fix.

### CODE REVIEW

File: xbmc/windowing/X11/WinSystemX11GLESContext.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #include "system.h"
 
-#if defined(HAVE_X11) && defined(HAS_EGL)
+#if defined(HAVE_X11) && defined(HAS_GLES)
```
Comment (Razzeee) [inline xbmc/windowing/X11/WinSystemX11GLESContext.cpp]: Did this commit slip in by accident?

**Comment by Razzeee** [comment]
This fixes the build failing, I did the same. But the test are still failing and need to be aligned to the new behavior with https://github.com/xbmc/xbmc/pull/8080

std::string DatabaseUtils::GetField(Field field, const MediaType &mediaType, DatabaseQueryPart queryPart) need to know about the multiple ratings that are now in place.

File: xbmc/windowing/X11/WinSystemX11GLESContext.cpp (hunk 1 above)
Comment (hudokkow) [inline xbmc/windowing/X11/WinSystemX11GLESContext.cpp]: No, not by accident. It doesn't build without it.
Thinking of it, it probably should be
<code>if defined(HAVE_X11) && defined(HAS_EGL) || defined(HAS_GLES)</code>

Comment (FernetMenta) [inline xbmc/windowing/X11/WinSystemX11GLESContext.cpp]: EGL is not GLES. builds for X11 require EGL. if this is not defines the build environment has an issue

File: xbmc/utils/test/TestDatabaseUtils.cpp [hunk 2]
```diff
@@ -704,7 +704,7 @@ TEST(TestDatabaseUtils, GetField_MediaTypeEpisode)
                                    DatabaseQueryPartSelect);
   EXPECT_STREQ(refstr.c_str(), varstr.c_str());
 
-  refstr = StringUtils::Format("episode_view.c%02d", VIDEODB_ID_EPISODE_RATING);
+  refstr = StringUtils::Format("episode_view.c%02d", VIDEODB_ID_EPISODE_RATING_ID);
```
Comment (phate89) [inline xbmc/utils/test/TestDatabaseUtils.cpp]: Now FieldRating use the value directly from the view named rating so refstr should be "episode_view.rating". The same for tv show, movie and musicvideo

File: xbmc/utils/test/TestDatabaseUtils.cpp [hunk 3]
```diff
@@ -477,12 +477,12 @@ TEST(TestDatabaseUtils, GetField_MediaTypeMovie)
...
   EXPECT_STREQ(refstr.c_str(), varstr.c_str());
 
-  refstr = StringUtils::Format("movie_view.c%02d", VIDEODB_ID_RATING);
+  refstr = StringUtils::Format("movie_view.c%02d", VIDEODB_ID_RATING_ID);
   varstr = DatabaseUtils::GetField(FieldRating, MediaTypeMovie,
                                    DatabaseQueryPartSelect);
   EXPECT_STREQ(refstr.c_str(), varstr.c_str());
 
-  refstr = StringUtils::Format("CAST(movie_view.c%02d as DECIMAL(5,3))", VIDEODB_ID_RATING);
+  refstr = StringUtils::Format("CAST(movie_view.c%02d as DECIMAL(5,3))", VIDEODB_ID_RATING_ID);
```
Comment (phate89) [inline xbmc/utils/test/TestDatabaseUtils.cpp]: Now we use a float field so there's no need for a special check with Order by so this check should be dropped

**Comment by Razzeee** [comment]
I would really like to get this in.

**Comment by FernetMenta** [comment]
3c39d8e is wrong

**Comment by phate89** [comment]
IF you want I can fix my own mess fixing myself the testsuite

**Comment by Razzeee** [comment]
@phate89 
go ahead please I don't think @hudokkow will hold a grudge if you do that :)

**Comment by Razzeee** [comment]
Closed in favor of https://github.com/xbmc/xbmc/pull/8835

**Comment by hudokkow** [comment]
Thanks!

**Comment by Razzeee** [comment]
It still fails using linux btw.

**Comment by phate89** [comment]
What test fails?

**Comment by Razzeee** [comment]
I don't think it's a test that fails, seems more to be compilation?
See http://jenkins.kodi.tv/view/Helpers/job/TestMulti-All/472/ when it finishes.

**Comment by hudokkow** [comment]
> No, not by accident. It doesn't build without it.

This is still true. It might be the wrong fix, but it builds with that commit.

**Comment by FernetMenta** [comment]
sorry, I was wrong. https://github.com/hudokkow/xbmc/commit/3c39d8ea0835a9837525083f690812c6f5e046f5 is correct

**Comment by fritsch** [comment]
There is still an EGL dependency in that code which is not covered by GLES

2016-01-12 9:10 GMT+01:00 Rainer Hochecker <notifications@github.com>:

> sorry, I was wrong. hudokkow@3c39d8e
> <https://github.com/hudokkow/xbmc/commit/3c39d8ea0835a9837525083f690812c6f5e046f5>
> is correct
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/8801#issuecomment-170833422>.
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B

**Comment by FernetMenta** [comment]
availability of GLES implies EGL too.

**Comment by fritsch** [comment]
Yes, you are right.

**Comment by hudokkow** [comment]
Okido! I'll PR just that commit.

**Pull request closed without merging**